packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -764,6 +868,44 @@ export class BasicCrawler<
             await this.sessionPool!.teardown();
         }
     }
+
+    protected _handlePropertyNameChange<New, Old>({
```

Comment:
this looks quite crazy :] maybe it would be easier to have multiple (but much simpler) methods, one for each deprecated property? that way you dont need that `CrawlerRenames` thingy at all

Reply:
I don't like to have those `CrawlerRenames` props, so that is what I would like to get rid of mostly.

Reply:
I've removed the `CrawlerRenames` in favor of an implementation similar to the one done in BrowserCrawler